opm/autodiff/NewtonSolver.hpp

```diff
@@ -66,7 +66,7 @@ namespace Opm {
         /// \param[in]      param   parameters controlling nonlinear Newton process
         /// \param[in, out] model   physical simulation model
         explicit NewtonSolver(const SolverParameters& param,
-                              PhysicalModel& model);
+                              std::shared_ptr<PhysicalModel> model);
```

Comment:
If you use a smart pointer as a parameter type, you're implicitly stating that you want to take part in the lifetime management of the underlying object.  Unless you want to do that--and there are certainly cases where you do want to do that--you should accept the parameter as a reference (typically) or a pointer and not care how the caller chooses to manage the object in question.

See also Sutter's [Back to Basics](https://www.youtube.com/watch?v=xnqTKD8uD64) talk from cppcon 2014 (1 hour, 40 minutes duration) about the same and related subjects.